wfe/wfe.go

```diff
@@ -2565,3 +2584,125 @@ func (wfe *WebFrontEndImpl) processRevocation(
 	})
 	return nil
 }
+
+// Verify the External Account Binding in the request and return the same JSON
+// object that was given in the request if successful. If no External Account
+// Binding was given then return nil however if it is required by the server
+// then error.
+func (wfe *WebFrontEndImpl) verifyEAB(newAcctReq newAccountRequest,
+	outerPostData *authenticatedPOST) (*acme.JSONSigned, *acme.ProblemDetails) {
+	if newAcctReq.ExternalAccountBinding == nil {
+		if wfe.requireEAB {
+			return nil, acme.ExternalAccountRequiredProblem(
+				"ACME server policy requires newAccount requests must include a value for the 'externalAccountBinding' field")
+		}
+
+		return nil, nil
+	}
+
+	//1.  Verify that the value of the field is a well-formed JWS
+	eabBytes, err := json.Marshal(newAcctReq.ExternalAccountBinding)
+	if err != nil {
+		return nil, acme.MalformedProblem(
```

Comment:
I think this case should be an internal err problem with a distinct error message from the case where `jose.ParseSigned` fails.

It seems to be that we've already unmarshaled this part of the request JSON from the raw outer JWS body. If there's an error marshaling it back to JSON something def. funky is going on. Am I overlooking something that makes this a better fit for a malformed problem? (If so I still think it needs a unique err message).